script/utils/rewards_testing/TestRewardsV2.sol

```diff
@@ -0,0 +1,334 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+import "./TransactionSubmitter.sol";
+import "script/utils/ExistingDeploymentParser.sol";
+
+import "@openzeppelin/contracts/token/ERC20/presets/ERC20PresetFixedSupply.sol";
+import "zeus-templates/utils/ZeusScript.sol";
+
+contract TestRewardsV2 is ZeusScript {
+    uint256 maxOperatorIndex = 80;
+    string MNEMONIC;
+    TransactionSubmitter transactionSubmitter;
+    string statePath = "script/utils/rewards_testing/preprod_state.json";
+
+
+    function tx_prep() public parseState {
+        // Deploy token 
+        (address[] memory operatorsRegisteredToAVS, uint256[] memory privateKeys) = _getAllOperatorsRegisteredToAVS(address(_rewardingServiceManager()));
+
+        // for the first 3/5 of the operators, set the AVS split to random values
+        for (uint256 i = 0; i < operatorsRegisteredToAVS.length*3/5; i++) {
+            vm.startBroadcast(privateKeys[i]);
+            _rewardsCoordinator().setOperatorAVSSplit(operatorsRegisteredToAVS[i], address(_rewardingServiceManager()), uint16(vm.randomUint(1, 10000)));
+            vm.stopBroadcast();
+        }
+
+        // for 1/5 through 4/5 of the operators, set the PI split to random values
+        for (uint256 i = operatorsRegisteredToAVS.length*1/5; i < operatorsRegisteredToAVS.length*4/5; i++) {
+            vm.startBroadcast(privateKeys[i]);
+            _rewardsCoordinator().setOperatorPISplit(operatorsRegisteredToAVS[i], uint16(vm.randomUint(1, 10000)));
+            vm.stopBroadcast();
+        }
+
+        // this leaves 1/5 of the operators with no splits set
+    }
+
+    function tx_1() public parseState {
```

Comment:
Dumb question. But how do we know which AVS to run this transaction as?

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/blob/88577757e91230f480879cfe1207d167a0272267/script/utils/rewards_testing/TestRewardsV2.sol#L243

just taking the first element